examples/widgets/slider_demo.py

```diff
@@ -3,65 +3,82 @@
 Slider
 ======
 
-Using the slider widget to control visual properties of your plot.
+In this example, sliders are used to control the frequency and amplitude of
+a sine wave.
 
-In this example, a slider is used to choose the frequency of a sine
-wave. You can control many continuously-varying properties of your plot in
-this way.
+See :doc:`/gallery/widgets/slider_snap_demo` for an example of having
+the ``Slider`` snap to discrete values.
+
+See :doc:`/gallery/widgets/range_slider` for an example of using
+a ``RangeSlider`` to define a range of values.
 """
 import numpy as np
 import matplotlib.pyplot as plt
-from matplotlib.widgets import Slider, Button, RadioButtons
+from matplotlib.widgets import Slider, Button
+
+
+# The parametrized function to be plotted
+def f(t, amplitude, frequency):
+    return amplitude * np.sin(2 * np.pi * frequency * t)
 
-fig, ax = plt.subplots()
-plt.subplots_adjust(left=0.25, bottom=0.25)
 t = np.arange(0.0, 1.0, 0.001)
-a0 = 5
-f0 = 3
-delta_f = 5.0
-s = a0 * np.sin(2 * np.pi * f0 * t)
-l, = plt.plot(t, s, lw=2)
-ax.margins(x=0)
 
-axcolor = 'lightgoldenrodyellow'
-axfreq = plt.axes([0.25, 0.1, 0.65, 0.03], facecolor=axcolor)
-axamp = plt.axes([0.25, 0.15, 0.65, 0.03], facecolor=axcolor)
+# Define initial parameters
+init_amplitude = 5
+init_frequency = 3
 
-sfreq = Slider(axfreq, 'Freq', 0.1, 30.0, valinit=f0, valstep=delta_f)
-samp = Slider(axamp, 'Amp', 0.1, 10.0, valinit=a0)
+# Create the figure and the line that we will manipulate
+fig, ax = plt.subplots()
+line, = plt.plot(t, f(t, init_amplitude, init_frequency), lw=2)
 
+axcolor = 'lightgoldenrodyellow'
+ax.margins(x=0)
+
+# adjust the main plot to make room for the sliders
+plt.subplots_adjust(left=0.25, bottom=0.25)
 
+# Make a horizontal slider to control the frequency.
+axfreq = plt.axes([0.25, 0.1, 0.65, 0.03], facecolor=axcolor)
+freq_slider = Slider(
+    ax=axfreq,
+    label='Frequency',
+    valmin=0.1,
+    valmax=30.0,
+    valinit=init_amplitude,
+)
+
+# Make a vertically oriented slider to control the amplitude
+axamp = plt.axes([0.1, 0.25, 0.0225, 0.63], facecolor=axcolor)
+amp_slider = Slider(
+    ax=axamp,
+    label="Amplitude",
+    valmin=0.1,
+    valmax=10.0,
```

Comment:
```suggestion
    valmin=0,
    valmax=10,
```
I think it's nicer to have the amplitude going all the way down to 0.